profiler/profile_test.go

```diff
@@ -7,202 +7,22 @@ package profiler
-					t.Run("enabled", func(t *testing.T) {
-						prof1 := test.Prof1.Protobuf()
-						prof2 := test.Prof2.Protobuf()
-						p, cleanup := deltaProfiler(prof1, prof2)
-						defer cleanup()
-						// first run, should produce the current profile twice (a bit
-						// awkward, but makes sense since we try to add delta profiles as an
-						// additional profile type to ease the transition)
-						profs, err := p.runProfile(profType)
-						require.NoError(t, err)
-						require.Equal(t, 1, len(profs))
-						require.Equal(t, "delta-"+profType.Filename(), profs[0].name)
-						require.Equal(t, prof1, profs[0].data)
-
-						// second run, should produce p1 profile and delta profile
-						profs, err = p.runProfile(profType)
-						require.NoError(t, err)
-						require.Equal(t, 1, len(profs))
-						require.Equal(t, "delta-"+profType.Filename(), profs[0].name)
-						require.Equal(t, test.WantDelta.String(), protobufToText(profs[0].data))
-
-						// check delta prof details like timestamps and duration
-						deltaProf, err := pprofile.ParseData(profs[0].data)
-						require.NoError(t, err)
-						require.Equal(t, test.Prof2.Time.UnixNano(), deltaProf.TimeNanos)
-						require.Equal(t, deltaPeriod.Nanoseconds(), deltaProf.DurationNanos)
-					})
-
-					t.Run("disabled", func(t *testing.T) {
-						prof1 := test.Prof1.Protobuf()
-						prof2 := test.Prof2.Protobuf()
-						p, cleanup := deltaProfiler(prof1, prof2, WithDeltaProfiles(false))
-						defer cleanup()
-
-						profs, err := p.runProfile(profType)
-						require.NoError(t, err)
-						require.Equal(t, 1, len(profs))
-					})
-				})
-			}
-		}
-	})
-
-	t.Run("cpu", func(t *testing.T) {
-		p, err := unstartedProfiler(CPUDuration(10 * time.Millisecond))
-		p.testHooks.startCPUProfile = func(w io.Writer) error {
-			_, err := w.Write([]byte("my-cpu-profile"))
-			return err
-		}
-		p.testHooks.stopCPUProfile = func() {}
-		require.NoError(t, err)
-		start := time.Now()
-		profs, err := p.runProfile(CPUProfile)
-		end := time.Now()
-		require.NoError(t, err)
-		assert.True(t, end.Sub(start) > 10*time.Millisecond)
-		assert.Equal(t, "cpu.pprof", profs[0].name)
-		assert.Equal(t, []byte("my-cpu-profile"), profs[0].data)
-	})
-
-	t.Run("goroutine", func(t *testing.T) {
-		p, err := unstartedProfiler()
-		p.testHooks.lookupProfile = func(name string, w io.Writer, _ int) error {
-			_, err := w.Write([]byte(name))
-			return err
-		}
-		require.NoError(t, err)
-		profs, err := p.runProfile(GoroutineProfile)
-		require.NoError(t, err)
-		assert.Equal(t, "goroutines.pprof", profs[0].name)
-		assert.Equal(t, []byte("goroutine"), profs[0].data)
-	})
+func Test_goroutineDebug2ToPprof_CrashSafety(t *testing.T) {
+	err := goroutineDebug2ToPprof(panicReader{}, ioutil.Discard, time.Time{})
+	require.NotNil(t, err)
+	require.Equal(t, "panic: 42", err.Error())
+}
 
-	t.Run("goroutinewait", func(t *testing.T) {
-		const sample = `
+func TestGoroutineDebug2ToPprof(t *testing.T) {
```

Comment:
Git's diff algorithm has kind of butchered this change. It might be easier to review this new test by viewing the file.